### DISCUSSION TITLE
force and energy predicted by deep potential

### DISCUSSION THREAD

**Discussion by sooyaaa233** [deepmodeling/deepmd-kit#2849] [Q&A]
![image](https://github.com/deepmodeling/deepmd-kit/assets/126163538/69bed65f-2159-4f49-b698-8e0e988f7a24)
![image](https://github.com/deepmodeling/deepmd-kit/assets/126163538/a7a8aa3f-1bab-4a2f-9a01-1f3dea30d9c7)
As shown in the figure, when I used the init data set to evaluate and train the potential function for 16 rounds, these situations occurred. What may be the problem? Thanks!

**Comment by Taixi-Feng** [comment]
Please provide your input files.

**Reply by sooyaaa233** [reply]
this is my param.json : 
{
     "type_map": ["Cs","Pb","Br"],
     "mass_map": [133,207,80],
     "init_data_prefix": "../",
     "init_data_sys": ["init-cubic/cubic222.poscar.01x01x01/02.md/sys-0008-0008-0024/deepmd",
                       "init-tetra/tetra222.poscar.01x01x01/02.md/sys-0008-0008-0024/deepmd",
                       "init-ortho-pnma/pnma211.poscar.01x01x01/02.md/sys-0024-0008-0008/deepmd",
                       "init-ortho-cmcm/cmcm111.poscar.01x01x01/02.md/sys-0008-0008-0024/deepmd"
                      ],
     "training_init_model":true,
     "training_iter0_model_path":["./iter.000000/00.train/000/graph.pb",
                                  "./iter.000000/00.train/001/graph.pb",
                                  "./iter.000000/00.train/002/graph.pb",
                                  "./iter.000000/00.train/003/graph.pb"],
     "sys_configs_prefix": "../",
     "sys_configs": [
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/0-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/0-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/0-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/0-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/1-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/1-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/1-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/1-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/2-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/2-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/2-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/2-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/3-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/3-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/3-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/3-5/*/POSCAR"],

         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/4-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/4-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/4-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/4-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/4-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/4-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/4-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/4-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/5-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/5-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/5-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/5-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/6-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/6-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/6-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/6-5/*/POSCAR"],

         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/7-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/7-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/7-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/7-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/7-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/7-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/7-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/7-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/8-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/8-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/8-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/8-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/9-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/9-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/9-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/9-5/*/POSCAR"],


         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/10-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/10-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/10-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/10-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/10-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/10-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/10-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/10-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/11-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/11-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/11-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/11-10/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cmcm/11-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/cubic/11-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/pnma/11-5/*/POSCAR"],
         ["/home/hsh/learnmd/study/1/dpgen/CsPbBr3/shujuji/tetra/11-5/*/POSCAR"]            


         
     ],
     "_comment": " that's all ",
     "numb_models": 4,
     "default_training_param": {
         "model": {
             "type_map": ["Cs","Pb","Br"],
             "descriptor": {
                 "type": "se_e2_a", 
                 "sel": [20,20,40],
                 "set_davg_zero" : true ,
                 "rcut_smth": 2.0,
                 "rcut": 6.0,
                 "neuron": [25,50,100],
                 "resnet_dt": false,
                 "axis_neuron": 16,
                 "seed": 3397109743
             },
             "fitting_net": {
                 "neuron": [240,240,240],
                 "resnet_dt": true,
                 "seed": 2766198549,
                 "_comment" : "that's all"

             }
         },
         "learning_rate": {
             "type": "exp", 
             "start_lr": 0.001,
             "stop_lr" : 1e-8,
             "decay_steps": 80000
         },
         "loss": {
            "type":"ener", 
             "start_pref_e": 0.02,
             "limit_pref_e": 2,
             "start_pref_f": 1000,
             "limit_pref_f": 1,
             "start_pref_v": 0.01,
             "limit_pref_v": 1
         },
         "training": {
            "training_data":{
                "systems":
                      [["init-cubic/cubic222.poscar.01x01x01/02.md/sys-0008-0008-0024/deepmd"],
                       ["init-tetra/tetra222.poscar.01x01x01/02.md/sys-0008-0008-0024/deepmd"],
                       ["init-ortho-pnma/pnma211.poscar.01x01x01/02.md/sys-0024-0008-0008/deepmd"],
                       ["init-ortho-cmcm/cmcm111.poscar.01x01x01/02.md/sys-0008-0008-0024/deepmd"]
                      ],
                "batch_size":"auto"

            },
             "numb_steps": 400000,
             "seed":2654632956,
             "disp_file": "lcurve.out",
             "disp_freq": 2000,
             "save_freq": 2000,
             "_comment": "that's all"
         }
     },
     "model_devi_dt": 0.002,
     "model_devi_skip": 0,
     "model_devi_f_trust_lo": 0.12,
     "model_devi_f_trust_hi": 0.30, 
     "model_devi_e_trust_lo": 10000000000.0,
     "model_devi_e_trust_hi": 10000000000.0,
     "model_devi_clean_traj": true,
     "model_devi_merge_traj": true, 
     "model_devi_jobs": [
         {"sys_idx": [0,1,2,3],"temps": [50,100,150,200],"press": [1],"trj_freq": 10,"nsteps": 500,"ensemble": "nvt","_idx": "00"},
         {"sys_idx": [4,5,6,7],"temps": [50,100,150,200],"press": [1],"trj_freq": 10,"nsteps": 1000,"ensemble": "nvt","_idx": "01"},
         {"sys_idx": [8,9,10,11],"temps": [50,100,150,200],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "02"},
         {"sys_idx": [12,13,14,15],"temps": [50,100,150,200],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "03"},

         {"sys_idx": [16,17,18,19],"temps": [250,300,350,400],"press": [1],"trj_freq": 10,"nsteps": 500,"ensemble": "nvt","_idx": "04"},
         {"sys_idx": [20,21,22,23],"temps": [250,300,350,400],"press": [1],"trj_freq": 10,"nsteps": 1000,"ensemble": "nvt","_idx": "05"},
         {"sys_idx": [24,25,26,27],"temps": [250,300,350,400],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "06"},
         {"sys_idx": [28,29,30,31],"temps": [250,300,350,400],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "07"},

         {"sys_idx": [32,33,34,35],"temps": [450,500,550,600],"press": [1],"trj_freq": 10,"nsteps": 500,"ensemble": "nvt","_idx": "08"},
         {"sys_idx": [36,37,38,39],"temps": [450,500,550,600],"press": [1],"trj_freq": 10,"nsteps": 1000,"ensemble": "nvt","_idx": "09"},
         {"sys_idx": [40,41,42,43],"temps": [450,500,550,600],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "10"},
         {"sys_idx": [44,45,46,47],"temps": [450,500,550,600],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "11"},

         {"sys_idx": [48,49,50,51],"temps": [650,700,750,800],"press": [1],"trj_freq": 10,"nsteps": 500,"ensemble": "nvt","_idx": "12"},
         {"sys_idx": [52,53,54,55],"temps": [650,700,750,800],"press": [1],"trj_freq": 10,"nsteps": 1000,"ensemble": "nvt","_idx": "13"},
         {"sys_idx": [56,57,58,59],"temps": [650,700,750,800],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "14"},
         {"sys_idx": [60,61,62,63],"temps": [650,700,750,800],"press": [1],"trj_freq": 10,"nsteps": 2000,"ensemble": "nvt","_idx": "15"}

         

        
     ],
     
     "fp_style": "vasp",
     "shuffle_poscar": false,
     "fp_task_max": 240,
     "fp_task_min": 5,
     "fp_accurate_threshold":0.99,
     "fp_pp_path": "./",
     "fp_pp_files": ["POTCAR-Cs","POTCAR-Pb","POTCAR-Br"],
     "fp_incar": "./INCAR"
}


and INCAR : 
# 计算能量和力
ISIF     =   2
ISYM     =   2

PREC     =   Normal
ALGO     =   Fast

ENCUT    =   400
EDIFF    =   1E-06

LREAL    =   Auto
LWAVE    =   .FALSE.
LCHARG   =   .FALSE.
ISMEAR   =   0
SIGMA    =   0.05

IBRION   =   -1
NSW      =   0
EDIFFG   =   -0.02

KGAMMA   =   .TRUE.
KSPACING =   0.4

NELM     =   300

**Reply by Taixi-Feng** [reply]
Based on my experience, my suggestion is:
1. Have you tried their effects on the parameters of the set training model?
2. The number of iterations of DP-GEN for each type of configuration is somewhat small.
3. Prioritize the implementation of configuration acquisition under the NPT ensemble
Statement:I am only a volunteer, so the following suggestions only represent me and not the Deep Potential developers.

**Reply by chtchelkatchev** [reply]
You have very large kspasing and so energy computed by vasp is incorrect. Reduce kspacing below 0.2 (check convergence!). I also had similar problem solved like that...

**Reply by sooyaaa233** [reply]
thank you ! I'll try it.

**Reply by sooyaaa233** [reply]
Now I know that the point far away from y=x in the third phase is a problem with vasp calculation, but for the other three phases, I still don’t know where the problem lies. They are all the same parameter training with different effects, which makes me very confused. . Regarding lowering kspacing, I have adjusted it to 0.2 a few days ago, but until the fifth round, the energy still shows such a deviation.
